crates/nu-parser/src/parser.rs

```diff
@@ -294,7 +294,7 @@ pub fn parse_external_call(
             let (arg, err) = parse_dollar_expr(working_set, *span, expand_aliases_denylist);
             error = error.or(err);
             args.push(arg);
-        } else if contents.starts_with(b"(") {
+        } else if contents.starts_with(b"[") {
```

Comment:
I think this might work (so you can pass `nu -c [ 'a' '-b' '--c' ]`, for example), but intstead of `parse_full_cell_path()`, I'd use `parse_list_expression()`.

I'm not sure why this conditional was there in the first place since it's already checked above. Maybe @jntrnr knows?

Reply:
Thank you :-) `parse_list_expression` is explicit and better.

> I'm not sure why this conditional was there in the first place since it's already checked above.

+1 for the not sure...

Reply:
I think the parens were originally so that you can do stuff like:

```
./external ("foo" + "bar.txt")
```

I think we still want to support this. If we bring in support for the string arrays, that should probably be a separate thing.

Reply:
@jntrnr  Isn't the subexpression checked on the line 293 though?

Reply:
hmmm, that 293 line looks odd. Are we really checking for `(` as the start and then calling `parse_dollar_expr`? Does this cause any bugs in the current system?

Reply:
`parse_dollar_expr` goes to `parse_full_cell_path` which checks for `(`. So to me it seems this line is a duplicate.

Reply:
I wonder if `parse_dollar_expr` just needs to be called something different now if it's doing more than parsing dollar expressions 😅

Reply:
Yeah, it doesn't seem very accurate anymore :-D. But I guess we can maybe rename it and leave the `[` check for the list parsing?

Reply:
Hmm... so I change to something like this to make the logic more explicit?
```rust
if contents.starts_with(b"$") {
    let (arg, err) = parse_dollar_expr(working_set, *span, expand_aliases_denylist);
    error = error.or(err);
    args.push(arg);
} else if contents.starts_with(b"(") {
     let (arg, err) =
         parse_full_cell_path(working_set, None, *span, expand_aliases_denylist);
     error = error.or(err);
     args.push(arg);
 }
```

Reply:
If you look at the `parse_dollar_expr` code it's actually parsing more than a dollar expression, so the original condition should be preserved. It's just a matter of adding for `[...]`.